tuplex/core/src/ee/local/Sort.cc

```diff
@@ -0,0 +1,605 @@
+        // gets the "right" row
+        tuplex::Row rRow = tuplex::Row::fromMemory(schema, partitionPtrs.at(r.first.first) + r.second.first,
+                                                   r.second.second - r.second.first);
+        // gets a list of the types for each column
+        std::vector<python::Type> colTypes = schema.getRowType().parameters();
+        int counter = 0;
+        while (counter != order.size()) {
+            // gets the current column to sort by
+            int currentColIndex = order.at(counter);
+            // gets the current type of this current column
+            python::Type colType = colTypes.at(currentColIndex);
+            // sorting for integer
+            counter++;
+            if (colType == python::Type::I64) {
+                int64_t lInt = lRow.getInt(currentColIndex);
+                int64_t rInt = rRow.getInt(currentColIndex);
+                if (lInt == rInt) {continue;} // edge case where there is no other column to sort by
+                else if (orderEnum.at(counter-1) == 3) {
+                    std::ostringstream o1;
+                    std::ostringstream o2;
+                    o1 << lInt;
+                    o2 << rInt;
+                    std::string lStrInt = o1.str();
+                    std::string rStrInt = o2.str();
+                    return lStrInt.length() < rStrInt.length();
+                }
+                else if (orderEnum.at(counter-1) == 4) {
+                    std::ostringstream o1;
+                    std::ostringstream o2;
+                    o1 << lInt;
+                    o2 << rInt;
+                    std::string lStrInt = o1.str();
+                    std::string rStrInt = o2.str();
+                    return lStrInt.length() > rStrInt.length();
+                }
+                else if (orderEnum.at(counter-1) == 5) {
+                    std::ostringstream o1;
+                    std::ostringstream o2;
+                    o1 << lInt;
+                    o2 << rInt;
+                    std::string lStrInt = o1.str();
+                    std::string rStrInt = o2.str();
+                    return lStrInt < rStrInt;
+                }
+                else if (orderEnum.at(counter-1) == 6) {
+                    std::ostringstream o1;
+                    std::ostringstream o2;
+                    o1 << lInt;
+                    o2 << rInt;
+                    std::string lStrInt = o1.str();
+                    std::string rStrInt = o2.str();
+                    return lStrInt > rStrInt;
+                }
+                else if (orderEnum.at(counter-1) == 2) {
+                    return lInt > rInt;
+                }
+                else {
+                    return lInt < rInt;
+                }
+                // sorting for floats
+            } else if (colType == python::Type::F64) {
+                double lDub = lRow.getDouble(currentColIndex);
```

Comment:
maybe add note. Comparing string size with float formats could lead to funny results. Let's try to find the right strategy here & describe here.